pandas/core/frame.py

```diff
@@ -1155,8 +1152,8 @@ def to_string(
         ...
 
     @Substitution(
-        header_type="bool or sequence of str",
-        header="Write out the column names. If a list of strings "
+        header_type="bool or array-like of column names",
```

Comment:
see my comment https://github.com/pandas-dev/pandas/issues/47215#issuecomment-1146773597 about the (imo) vague definition of array-like. what about list-like?

Reply:
Well, we do accept a `Series` or `Index` here, which I guess is "list-like", so I could do that instead.

Reply:
> Well, we do accept a `Series` or `Index` here, which I guess is "list-like", so I could do that instead.

Turns out we don't accept a `Series` or `Index`, but do accept a numpy array, but I think we're better off documenting it as a list, and restricting it to a `list[str]` .  It does have to be a list of strings.